Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -5,99 +5,121 @@ Trace-VstsEnteringInvocation $MyInvocation
-
-	#### MAIN EXECUTION OF AZURERM WEBAPP DEPLOYMENT TASK BEGINS HERE ####
-
-	# Get msdeploy.exe path
-	$msDeployExePath = Get-MsDeployExePath
+    # Get inputs.
+    $WebAppName = Get-VstsInput -Name WebAppName -Require
+    $DeployToSlotFlag = Get-VstsInput -Name DeployToSlotFlag -Require -AsBool
+    $ResourceGroupName = Get-VstsInput -Name ResourceGroupName
+    $SlotName = Get-VstsInput -Name SlotName
+    $Package = Get-VstsInput -Name Package -Require
+    $RemoveAdditionalFilesFlag = Get-VstsInput -Name RemoveAdditionalFilesFlag -AsBool
+    $ExcludeFilesFromAppDataFlag = Get-VstsInput -Name ExcludeFilesFromAppDataFlag -AsBool
+    $TakeAppOfflineFlag = Get-VstsInput -Name TakeAppOfflineFlag -AsBool
+    $VirtualApplication = Get-VstsInput -Name VirtualApplication 
+    $AdditionalArguments = Get-VstsInput -Name AdditionalArguments 
+    $WebAppUri = Get-VstsInput -Name WebAppUri
+    $SetParametersFile = Get-VstsInput -Name SetParametersFile
+    $XmlTransformation = Get-VstsInput -Name XmlTransformation
+    $XdtFilesRoot = Get-VstsInput -Name XdtFilesRoot
+
+    # Initialize Azure.
+
+    Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
+    Initialize-Azure
+
+    # Import the loc strings.
+    Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
+
+    $WebAppUri = $WebAppUri.Trim()
+    $Package = "$Package".Trim('"').Trim()
+
+    if( [string]::IsNullOrEmpty($Package) ){
+        Throw (Get-VstsLocString -Key "Invalidwebapppackagepathprovided")
+    }
 
-	# Ensure that at most a package (.zip) file is found
-	$packageFilePath = Get-SingleFilePath -file $Package
+    $SetParametersFile = "$SetParametersFile".Trim('"').Trim()
+
+
+    # Load all dependent files for execution
+    . $PSScriptRoot/AzureUtility.ps1
+    . $PSScriptRoot/Utility.ps1
+    . $PSScriptRoot/FindInstalledMSDeploy.ps1
+    . $PSScriptRoot/CompressionUtility.ps1
+	. $PSScriptRoot/XdtTransformation.ps1
+
+    # Importing required version of azure cmdlets according to azureps installed on machine
+    $azureUtility = Get-AzureUtility
+
+    Write-Verbose  "Loading $azureUtility"
+    . $PSScriptRoot/$azureUtility -Force
+
+    #### MAIN EXECUTION OF AZURERM WEBAPP DEPLOYMENT TASK BEGINS HERE ####
+
+    # Get msdeploy.exe path
+    $msDeployExePath = Get-MsDeployExePath
+
+    # Ensure that at most a package (.zip) file is found
+    $packageFilePath = Get-SingleFilePath -file $Package
+
+    if($XmlTransformation -eq "true")
+    {
+        # Unzip the source package
+        $unzippedPath = UnzipWebDeployPkg -PackagePath $packageFilePath
+        # Search for all the web.config files
+        $webconfigFiles = Find-VstsFiles -LegacyPattern "$unzippedPath\**\web.config" -IncludeFiles
+        # Foreach web.config file apply Web.Release.Config and Web.Environment.config
+        foreach ($configFile in $webconfigFiles) {
+            FindAndApplyTransformation -baseFile $configFile -tranformFile "web.release.config" -xdtFilesRoot $XdtFilesRoot
```

Comment:
Ambrish wanted *.release.config to be applied by default. I asked for clarification he said that is the way usually users will do.